### PR TITLE
chore: make Java Engine pass specs

### PR DESCRIPTION
**Pull request by gastonfournier**
## About the changes
Adapt Java Engine to latest changes in Yggdrasil and make tests pass the specification

### CODE REVIEW

File: java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java [hunk 1]
```diff
@@ -25,40 +26,62 @@ interface UnleashFFI extends Library {
 
     Pointer take_state(Pointer ptr, String toggles);
 
-    boolean check_enabled(Pointer ptr, String name, String context);
+    Pointer check_enabled(Pointer ptr, String name, String context);
 
     Pointer check_variant(Pointer ptr, String name, String context);
```
Comment (sighphyre) [inline java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java:32]: We'll need the metrics methods here too but that can be in another PR

**Review by sighphyre — APPROVED** [review]
Yeah this is really nice, the error handling is great

File: java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java (hunk 1 above)
Comment (gastonfournier) [reply java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java:32]: Yes, will cover that here: https://linear.app/unleash/issue/2-1560/implement-metrics
